internal/server/auditsink/auditsink.go

```diff
@@ -0,0 +1,97 @@
+package auditsink
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	"go.uber.org/zap"
+)
+
+// AuditEvent holds information that represents an audit internally.
+type AuditEvent struct {
+	ResourceName string `json:"resourceName"`
+}
+
+// AuditSink is the abstraction for various audit sink configurations
+// that Flipt will support.
+type AuditSink interface {
+	SendAudits([]AuditEvent) error
+	fmt.Stringer
+}
+
+// Publisher holds information about the configured sinks that we are going
+// to send audit events to.
+type Publisher struct {
+	mtx         sync.Mutex
+	logger      *zap.Logger
+	bufferSize  int
+	sinks       []AuditSink
+	auditEvents []AuditEvent
+	ticker      *time.Ticker
+}
+
+// NewPublisher is the constructor for a Publisher.
+func NewPublisher(logger *zap.Logger, bufferSize int, sinks []AuditSink, tickerDuration time.Duration) *Publisher {
+	p := &Publisher{
+		logger:      logger,
+		bufferSize:  bufferSize,
+		sinks:       sinks,
+		auditEvents: make([]AuditEvent, 0),
```

Comment:
You could pre-allocate the capacity here to avoid ever having to grow the buffer.
```suggestion
		auditEvents: make([]AuditEvent, 0, bufferSize),
```

Reply:
@GeorgeMac Yeah meant to circle back to this. Will add the capacity there to avoid unnecessary buffer growth 👍